### ISSUE TITLE
If a mod that doesn't specify an icon is present, the log can be spammed

### ISSUE THREAD

**Issue by altrisi** [Prospector/ModMenu#363]
The fabric.mod.json spec specifies that `icon` is an optional field, therefore, not specifying an icon should be fine.

However, if a mod doesn't contain the `icon` key in the fabric.mod.json file, ModMenu will print a warning to the log not only once or when opening the menu, but also every time a mod that can be collapsed is collapsed or expanded (it doesn't have to be the "faulty" one).

```log
[14:25:16] [Render thread/WARN] (Mod Menu) Warning! Mod modid has a broken icon, loading default icon
```

**Comment by jackassmc** [comment]
Thank you for this bug report! This will be fixed in a future version of ModMenu.

**Comment by MMK21Hub** [comment]
Note that this also spams the log with warnings when resizing the window. My guess is that `ModsScreen.render()` is called whenever the window size changes.

**Comment by jackassmc** [comment]
This will be fixed in the next release.

**Comment by TheMrEngMan** [comment]
This can still happen with mods that don't have an icon file, but have one defined in their `fabric.mod.json` or define an incorrect path to the icon file.
Perhaps if the exception is anything other than `NoSuchFileException`, then `LOGGER.error` should be used [here](https://github.com/TerraformersMC/ModMenu/blob/1.19.4/src/main/java/com/terraformersmc/modmenu/util/mod/fabric/FabricIconHandler.java#L44) as it is now, otherwise, silently ignore the error since a warning will still be printed [here](https://github.com/TerraformersMC/ModMenu/blob/1.19.4/src/main/java/com/terraformersmc/modmenu/util/mod/fabric/FabricMod.java#L162), which is sufficient to diagnose which mod has the icon issue while avoiding log error spam.

In other words, I am proposing changing the catch block [here](https://github.com/TerraformersMC/ModMenu/blob/1.19.4/src/main/java/com/terraformersmc/modmenu/util/mod/fabric/FabricIconHandler.java#L42) to:
```java
if(!(t instanceof NoSuchFileException)) {
    LOGGER.error("Invalid mod icon for icon source {}: {}", iconSource.getMetadata().getId(), iconPath, t);
}  
// Otherwise, the current mod does not properly define it's icon file, 
// resort to using the default (warning will be printed from calling method)
return null;
```
